Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp

```diff
@@ -823,7 +828,23 @@ class CombinatorialKalmanFilter {
         currentBranch.tipIndex() = currentTip;
         auto currentState = currentBranch.outermostTrackState();
         if (expectMeasurements) {
-          currentBranch.nHoles()++;
+           static const BoundaryTolerance exclude_sensor_border
+              = BoundaryTolerance(BoundaryTolerance::AbsoluteCartesian{-2*UnitConstants::mm,-2*UnitConstants::mm});
+           if (currentState.referenceSurface().insideBounds(currentState.parameters().template head<2>(),
+                                                            exclude_sensor_border)) {
+              currentBranch.nHoles()++;
+           }
+           else {
+            if (!keepEdgeHoles) {
+              currentBranch.nHoles()++;
+            } 
+            else {
+              auto typeFlags = currentState.typeFlags();
+              typeFlags.reset(TrackStateFlag::HoleFlag);
+              typeFlags.set(TrackStateFlag::EdgeHoleFlag);
+              //currentBranch.nEdgeHoles()++;
+            }
+           }
```

Comment:
IMO this is an experiment choice and it should not be part of the Core CKF. This can be put into the track state creator to put the user in control

Reply:
> IMO this is an experiment choice and it should not be part of the Core CKF. This can be put into the track state creator to put the user in control

Not sure whether there is a universally accepted definition of what a hole is.  Is it really a how if the edge is within the uncertainties of the trajectory ? It is a definition, but I guess one can argue either way, whether it should be counted as a hole or not. Sure an ad hoc  2mm tolerance is not universal.